asm-utils/src/data_storage.rs

```diff
@@ -7,23 +7,23 @@ use crate::{
     utils::{alignment_size, next_aligned},
 };
 
-pub enum SingleDataValue<'a> {
+pub enum SingleDataValue {
```

Comment:
could be nice to have a small comment on why this type exists and a comment on each variant about which situations they come from

Reply:
Done.